File: cpp/pangram/pangram.cpp
#include "pangram.h"

#include <cctype>
#include <set>

namespace pangram
{
bool is_pangram(std::string phrase)
{
    std::set<char> letters;
    for (auto c : phrase)
    {
        if (!std::isalpha(c))
        {
            continue;
        }

        letters.insert(std::tolower(c));
    }

    return letters.size() == 26;
}
} // namespace pangram

File: zig/pangram/pangram.zig
const std = @import("std");

pub fn isPangram(str: []const u8) bool {
    var seen = [_]u8{0} ** 26;
    for (str) |c| {
        if ((c >= 'a' and c <= 'z') or (c >= 'A' and c <= 'Z')) {
            const index = std.ascii.toLower(c) - 'a';
            seen[index] += 1;
        }
    }
    
    for (seen) |count| {
        if (count == 0) return false;
    }
    return true;
}
